app.controllers.js

```diff
@@ -16,14 +16,27 @@ const getArticles = (req, res) => {
 
 const getArticleById = (req, res, next) => {
     const {article_id} = req.params;
-
     fetchArticleById(article_id).then((article) => {
         return article
     })
     .then((article) => {
         res.status(200).send({ article });
     })
-    .catch(next);
+    .catch((err) => {
```

Comment:
These errors never reach your middleware because you are invoking .status and .send in your controller.

Use next() to pass your custom errors to the error handling middleware in your app.js.